base/promotion.jl

```diff
@@ -223,21 +223,22 @@ minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 _default_type(T::Type) = (@_pure_meta; T)
 
 if isdefined(Core, :Inference)
-    _promote_op(f::ANY, t::ANY) = Core.Inference.return_type(f, t)
+    _return_type(f::ANY, t::ANY) = Core.Inference.return_type(f, t)
 else
-    _promote_op(f::ANY, t::ANY) = Any
+    _return_type(f::ANY, t::ANY) = Any
 end
 
-promote_op(::Any...) = (@_pure_meta; Any)
```

Comment:
Removing this fallback seems like a step forward (for safety) though somewhat orthogonal to the other changes in this pull request. I imagine this removal impacts other aspects of this pull request in ways I do not immediately see? Thanks!